.github/workflows/build-and-test-device.yaml

```diff
@@ -50,27 +50,12 @@ jobs:
       NANOARROW_ARROW_TESTING_DIR: '${{ github.workspace }}/arrow-testing'
 
     steps:
-      - name: Checkout repo
-        uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-
-      - name: Checkout arrow-testing
-        uses: actions/checkout@v3
-        with:
-          repository: apache/arrow-testing
-          fetch-depth: 0
-          path: arrow-testing
+      - uses: actions/checkout@v4
 
-      - name: Install dependencies
+      - name: Install memcheck dependencies
+        if: matrix.config.label == 'default-build'
         run: |
-          sudo apt-get update
-          sudo apt install -y -V ca-certificates lsb-release wget cmake valgrind
-          wget https://apache.jfrog.io/artifactory/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
-          sudo apt-get install -y -V ./apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb
-          sudo apt-get update
-          sudo apt-get install -y -V libarrow-dev
-          rm apache-arrow-apt-*.deb
```

Comment:
You are removing the installation of arrow here but I don't see the cache/build being added? Was it not needed?